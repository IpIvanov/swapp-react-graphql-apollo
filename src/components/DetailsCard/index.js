import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import { Card, CardMedia, CardContent, Typography, makeStyles } from '@material-ui/core';
import styles from './styles';
import ThemeContext from '../../contexts/ThemeContext';

const DetailsCard = ({ title, image, details }) => {
  const {
    theme,
  } = useContext(ThemeContext);
  const useStyles = makeStyles(styles);
  const classes = useStyles({ theme });

  return (
    <Card className={classes.card}>
      <Typography gutterBottom variant="h5" component="h2" className={classes.title}>
        {title}
      </Typography>
      <CardMedia
        className={classes.cover}
        image={image}
        title={title}
      />
      <CardContent className={classes.content}>
        {details.map((detail) => (
          <Typography
            variant="body2"
            color="textSecondary"
            component="p"
            className={classes.description}
            align="left"
            key={detail.key}
          >
            {`${detail.key}: `}
            <span className={classes.detailValue}>
              {detail.value}
              {' '}
              {detail.key === 'Cost' && 'credits'}
            </span>
          </Typography>
        ))}
      </CardContent>
    </Card>
  );
};

DetailsCard.propTypes = {
  title: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  details: PropTypes.array.isRequired,
};

export default DetailsCard;
