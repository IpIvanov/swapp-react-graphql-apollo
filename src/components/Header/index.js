import React, { useContext } from 'react';
import { useHistory } from 'react-router-dom';
import { useApolloClient } from '@apollo/react-hooks';
import { Input } from '@material-ui/icons';
import { makeStyles, Typography, Link } from '@material-ui/core';
import styles from './styles';
import ThemeContext from '../../contexts/ThemeContext';

const Header = () => {
  const {
    theme,
    setTheme,
  } = useContext(ThemeContext);
  const useStyles = makeStyles(styles);
  const classes = useStyles({ theme });
  const history = useHistory();
  const client = useApolloClient();

  const linkTo = (e, route) => {
    e.preventDefault();
    history.push(route);
  };

  const logOut = () => {
    client.writeData({ data: { authenticated: false } });
    history.push('/');
  };

  const toggleTheme = () => {
    const themeStorage = localStorage.getItem('theme') === 'light' ? 'dark' : 'light';
    setTheme(themeStorage);
    localStorage.setItem('theme', themeStorage);
  };

  return (
    <div className={`${classes.container}`}>
      <Typography
        className={classes.linkHome}
        onClick={toggleTheme}
      >
        SWAPP
      </Typography>
      <div className={classes.rightHeader}>
        <Link
          href="/"
          className={classes.link}
          underline="none"
          onClick={(e) => {
            linkTo(e, '/episodes');
          }}
        >
          Episodes
        </Link>
        <Link
          href="/"
          className={classes.link}
          underline="none"
          onClick={(e) => {
            linkTo(e, '/characters');
          }}
        >
          Characters
        </Link>
        <div className={classes.exit}>
          <Input
            onClick={logOut}
          />
        </div>
      </div>
    </div>
  );
};

export default Header;
