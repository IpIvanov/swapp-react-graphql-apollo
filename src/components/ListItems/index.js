import React from 'react';
import { Link } from 'react-router-dom';
import { makeStyles, Typography, Card, Grid, Box, Button, CardMedia, CardContent } from '@material-ui/core';
import styles from './styles';

const ListItems = ({ listItems, loadMoreIsVisible, loadMoreHandler, linkTo, mdColumns }) => {
  const useStyles = makeStyles(styles);
  const classes = useStyles();

  return (
    <Grid container display="flex" direction="row">
      <Grid container spacing={2}>
        {listItems.map(({ name, id, image }, i) => (
          <Grid item xs={12} md={mdColumns} key={i.toString()}>
            <Link to={`/${linkTo}/${id}`} className={classes.linkTo}>
              <Card className={classes.card}>
                <CardMedia
                  className={classes.cover}
                  image={image}
                  title={name}
                  src={name}
                />
                <CardContent className={classes.content}>
                  <Typography component="h5" variant="h5" className={classes.name}>
                    {name}
                  </Typography>
                </CardContent>
              </Card>
            </Link>
          </Grid>
        ))}
      </Grid>
      {loadMoreIsVisible && (
        <Box display="flex" justifyContent="center" alignItems="center" flex="1">
          <Button
            color="primary"
            variant="outlined"
            size="small"
            className={classes.button}
            onClick={loadMoreHandler}
          >
            Load More
          </Button>
        </Box>
      )}
    </Grid>
  );
};

export default ListItems;
