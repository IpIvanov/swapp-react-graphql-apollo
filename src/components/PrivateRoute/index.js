/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { Redirect, Route } from 'react-router-dom';

const PrivateRoute = ({ component: Component, publicRoute, isAuthorized, ...rest }) => (
  <Route
    {...rest}
    render={(props) => (
      isAuthorized ? (<Component {...props} />)
        : (<Redirect to={{ pathname: publicRoute }} />)
    )}
  />
);

export default PrivateRoute;
