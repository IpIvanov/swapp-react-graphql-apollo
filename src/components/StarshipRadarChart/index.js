/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import PropTypes from 'prop-types';
import { RadarChart, CircularGridLines } from 'react-vis';

const StarshipRadarChart = (
  { starshipData: { starshipClass, maxAtmosphericSpeed, cost, crew, hyperdriveRating, maxMLPerHour },
    allStarshipsData },
) => {
  const data = allStarshipsData.filter((starship) => starship.node.starshipClass === starshipClass)
    .map((
      { node: { cost: Cost, crew: Crew, hyperdriveRating: hdRating, maxMLPerHour: maxMlPHour, maxAtmosphericSpeed: maSpeed } },
    ) => ({ Cost, Crew, 'HyperD Rat.': hdRating, 'Max ML/h': maxMlPHour, 'Max Atm. Speed': maSpeed }));
  let DOMAIN = [
    { name: 'Max Atm. Speed', domain: [0, Math.max(...data.map((o) => o['Max Atm. Speed']), 0)], tickFormat: (t) => t },
    { name: 'Cost', domain: [0, Math.max(...data.map((o) => o.Cost), 0)] },
    { name: 'Crew', domain: [0, Math.max(...data.map((o) => o.Crew), 0)] },
    { name: 'HyperD Rat.', domain: [0, Math.max(...data.map((o) => o['HyperD Rat.']), 0) + 100] },
    { name: 'Max ML/h', domain: [0, Math.max(...data.map((o) => o['Max ML/h']), 0)] },
  ];

  const dataObject = () => {
    const obj = {};
    if (maxAtmosphericSpeed) {
      obj['Max Atm. Speed'] = maxAtmosphericSpeed;
    }
    if (cost) {
      obj.Cost = cost;
    }
    if (crew) {
      obj.Crew = crew;
    }
    if (hyperdriveRating) {
      obj['HyperD Rat.'] = hyperdriveRating;
    }
    if (maxMLPerHour) {
      obj['Max ML/h'] = maxMLPerHour;
    }

    return obj;
  };

  const DATA = [
    dataObject(),
  ];

  DOMAIN = DOMAIN.map((item) => Object.keys(DATA[0]).map((title) => {
    if (title === item.name) {
      return item;
    }

    return undefined;
  })).flat().filter((item) => item !== undefined);

  return (
    <RadarChart
      animation
      data={DATA}
      domains={DOMAIN}
      style={{
        polygons: {
          fillOpacity: 0.4,
          strokeWidth: 2,
          strokeOpacity: 1,
        },
        axes: {
          text: {
            opacity: 1,
          },
          line: {
            strokeWidth: 2,
          },
        },
        labels: {
          textAnchor: 'middle',
          fontSize: 12,
          fontWeight: 600,
          fill: '#fff',
        },
      }}
      margin={{
        left: 50,
        top: 40,
        right: 50,
      }}
      width={400}
      height={400}
    >
      <CircularGridLines
        style={{ stroke: '#333' }}
        tickValues={[...new Array(10)].map((v, i) => i / 10 - 1)}
      />
    </RadarChart>
  );
};

StarshipRadarChart.propTypes = {
  starshipData: PropTypes.object.isRequired,
  allStarshipsData: PropTypes.array.isRequired,
};

export default StarshipRadarChart;
