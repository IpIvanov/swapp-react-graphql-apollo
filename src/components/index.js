import Header from './Header';
import PrivateRoute from './PrivateRoute';
import ListItems from './ListItems';
import DetailsCard from './DetailsCard';

export { Header, PrivateRoute, ListItems, DetailsCard };
