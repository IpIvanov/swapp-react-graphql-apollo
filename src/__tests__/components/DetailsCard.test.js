/* eslint global-require: 0 */
import React from 'react';
import { mount } from 'enzyme';
import ThemeContext from '../../contexts/ThemeContext';
import { DetailsCard } from '../../components';

describe('Components | DetailsCard', () => {
  it('renders without crashing', () => {
    const wrapper = mount(
      <ThemeContext.Provider value={{ theme: 'dark', setTheme: () => { } }}>
        <DetailsCard
          title="Test title"
          image="test-image.jpeg"
          details={
            [{
              key: 'test-key',
              value: 'test-value',
            }]
          }
        />
      </ThemeContext.Provider>,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
