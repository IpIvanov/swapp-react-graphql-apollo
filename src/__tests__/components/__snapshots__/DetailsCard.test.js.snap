// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components | DetailsCard renders without crashing 1`] = `
<WithStyles(ForwardRef(Card))
  className="makeStyles-card-1"
>
  <ForwardRef(Card)
    className="makeStyles-card-1"
    classes={
      Object {
        "root": "MuiCard-root",
      }
    }
  >
    <WithStyles(ForwardRef(Paper))
      className="MuiCard-root makeStyles-card-1"
      elevation={1}
    >
      <ForwardRef(Paper)
        className="MuiCard-root makeStyles-card-1"
        classes={
          Object {
            "elevation0": "MuiPaper-elevation0",
            "elevation1": "MuiPaper-elevation1",
            "elevation10": "MuiPaper-elevation10",
            "elevation11": "MuiPaper-elevation11",
            "elevation12": "MuiPaper-elevation12",
            "elevation13": "MuiPaper-elevation13",
            "elevation14": "MuiPaper-elevation14",
            "elevation15": "MuiPaper-elevation15",
            "elevation16": "MuiPaper-elevation16",
            "elevation17": "MuiPaper-elevation17",
            "elevation18": "MuiPaper-elevation18",
            "elevation19": "MuiPaper-elevation19",
            "elevation2": "MuiPaper-elevation2",
            "elevation20": "MuiPaper-elevation20",
            "elevation21": "MuiPaper-elevation21",
            "elevation22": "MuiPaper-elevation22",
            "elevation23": "MuiPaper-elevation23",
            "elevation24": "MuiPaper-elevation24",
            "elevation3": "MuiPaper-elevation3",
            "elevation4": "MuiPaper-elevation4",
            "elevation5": "MuiPaper-elevation5",
            "elevation6": "MuiPaper-elevation6",
            "elevation7": "MuiPaper-elevation7",
            "elevation8": "MuiPaper-elevation8",
            "elevation9": "MuiPaper-elevation9",
            "root": "MuiPaper-root",
            "rounded": "MuiPaper-rounded",
          }
        }
        elevation={1}
      >
        <div
          className="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-card-1 MuiPaper-rounded"
        >
          <WithStyles(ForwardRef(Typography))
            className="makeStyles-title-5"
            component="h2"
            gutterBottom={true}
            variant="h5"
          >
            <ForwardRef(Typography)
              className="makeStyles-title-5"
              classes={
                Object {
                  "alignCenter": "MuiTypography-alignCenter",
                  "alignJustify": "MuiTypography-alignJustify",
                  "alignLeft": "MuiTypography-alignLeft",
                  "alignRight": "MuiTypography-alignRight",
                  "body1": "MuiTypography-body1",
                  "body2": "MuiTypography-body2",
                  "button": "MuiTypography-button",
                  "caption": "MuiTypography-caption",
                  "colorError": "MuiTypography-colorError",
                  "colorInherit": "MuiTypography-colorInherit",
                  "colorPrimary": "MuiTypography-colorPrimary",
                  "colorSecondary": "MuiTypography-colorSecondary",
                  "colorTextPrimary": "MuiTypography-colorTextPrimary",
                  "colorTextSecondary": "MuiTypography-colorTextSecondary",
                  "displayBlock": "MuiTypography-displayBlock",
                  "displayInline": "MuiTypography-displayInline",
                  "gutterBottom": "MuiTypography-gutterBottom",
                  "h1": "MuiTypography-h1",
                  "h2": "MuiTypography-h2",
                  "h3": "MuiTypography-h3",
                  "h4": "MuiTypography-h4",
                  "h5": "MuiTypography-h5",
                  "h6": "MuiTypography-h6",
                  "noWrap": "MuiTypography-noWrap",
                  "overline": "MuiTypography-overline",
                  "paragraph": "MuiTypography-paragraph",
                  "root": "MuiTypography-root",
                  "srOnly": "MuiTypography-srOnly",
                  "subtitle1": "MuiTypography-subtitle1",
                  "subtitle2": "MuiTypography-subtitle2",
                }
              }
              component="h2"
              gutterBottom={true}
              variant="h5"
            >
              <h2
                className="MuiTypography-root makeStyles-title-5 MuiTypography-h5 MuiTypography-gutterBottom"
              >
                Test title
              </h2>
            </ForwardRef(Typography)>
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(CardMedia))
            className="makeStyles-cover-4"
            image="test-image.jpeg"
            title="Test title"
          >
            <ForwardRef(CardMedia)
              className="makeStyles-cover-4"
              classes={
                Object {
                  "img": "MuiCardMedia-img",
                  "media": "MuiCardMedia-media",
                  "root": "MuiCardMedia-root",
                }
              }
              image="test-image.jpeg"
              title="Test title"
            >
              <div
                className="MuiCardMedia-root makeStyles-cover-4"
                style={
                  Object {
                    "backgroundImage": "url(\\"test-image.jpeg\\")",
                  }
                }
                title="Test title"
              />
            </ForwardRef(CardMedia)>
          </WithStyles(ForwardRef(CardMedia))>
          <WithStyles(ForwardRef(CardContent))
            className="makeStyles-content-2"
          >
            <ForwardRef(CardContent)
              className="makeStyles-content-2"
              classes={
                Object {
                  "root": "MuiCardContent-root",
                }
              }
            >
              <div
                className="MuiCardContent-root makeStyles-content-2"
              >
                <WithStyles(ForwardRef(Typography))
                  align="left"
                  className="makeStyles-description-3"
                  color="textSecondary"
                  component="p"
                  key="test-key"
                  variant="body2"
                >
                  <ForwardRef(Typography)
                    align="left"
                    className="makeStyles-description-3"
                    classes={
                      Object {
                        "alignCenter": "MuiTypography-alignCenter",
                        "alignJustify": "MuiTypography-alignJustify",
                        "alignLeft": "MuiTypography-alignLeft",
                        "alignRight": "MuiTypography-alignRight",
                        "body1": "MuiTypography-body1",
                        "body2": "MuiTypography-body2",
                        "button": "MuiTypography-button",
                        "caption": "MuiTypography-caption",
                        "colorError": "MuiTypography-colorError",
                        "colorInherit": "MuiTypography-colorInherit",
                        "colorPrimary": "MuiTypography-colorPrimary",
                        "colorSecondary": "MuiTypography-colorSecondary",
                        "colorTextPrimary": "MuiTypography-colorTextPrimary",
                        "colorTextSecondary": "MuiTypography-colorTextSecondary",
                        "displayBlock": "MuiTypography-displayBlock",
                        "displayInline": "MuiTypography-displayInline",
                        "gutterBottom": "MuiTypography-gutterBottom",
                        "h1": "MuiTypography-h1",
                        "h2": "MuiTypography-h2",
                        "h3": "MuiTypography-h3",
                        "h4": "MuiTypography-h4",
                        "h5": "MuiTypography-h5",
                        "h6": "MuiTypography-h6",
                        "noWrap": "MuiTypography-noWrap",
                        "overline": "MuiTypography-overline",
                        "paragraph": "MuiTypography-paragraph",
                        "root": "MuiTypography-root",
                        "srOnly": "MuiTypography-srOnly",
                        "subtitle1": "MuiTypography-subtitle1",
                        "subtitle2": "MuiTypography-subtitle2",
                      }
                    }
                    color="textSecondary"
                    component="p"
                    variant="body2"
                  >
                    <p
                      className="MuiTypography-root makeStyles-description-3 MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignLeft"
                    >
                      test-key 
                      <span
                        className="makeStyles-detailValue-6"
                      >
                        test-value
                      </span>
                    </p>
                  </ForwardRef(Typography)>
                </WithStyles(ForwardRef(Typography))>
              </div>
            </ForwardRef(CardContent)>
          </WithStyles(ForwardRef(CardContent))>
        </div>
      </ForwardRef(Paper)>
    </WithStyles(ForwardRef(Paper))>
  </ForwardRef(Card)>
</WithStyles(ForwardRef(Card))>
`;
